visible.lua

```diff
@@ -0,0 +1,156 @@
+local LIBRARY_PATH = (...):match("(.-)[^%.]+$")
+local vector = require(LIBRARY_PATH .. "vector")
+local visible = {}
+
+
+local function comp(a, b, center)
+  if a.start == b.start then
+    if a.stop == nil then return false end
+    if b.stop == nil then return true end
+    return vector.alignment(a.start, a.stop, b.stop) > 0
+  end
+  return vector.polar_lt(a.start, b.start, center)
+end
+
+function visible.polygon(segments, center)
```

Comment:
The tricky thing with asking for a Vector object here is that it then is compatible only with Vector and not vector-light. People have their reasons for preferring vector-light. You could make it receive two arguments for center: center_or_x and y. Then you can interperet center_or_x as a vector if y is nil, as the x component of a vector from vector-light if y is not nil.